android/tangram/src/com/mapzen/tangram/MapController.java

```diff
@@ -737,6 +765,60 @@ void checkPointer(long ptr) {
         }
     }
 
+    void checkId(long id) {
+        if (id <= 0) {
+            throw new RuntimeException("Tried to perform an operation on an invalid id! This means you may have used an object that has been disposed and is no longer valid.");
```

Comment:
I'm not sure that we should apply the same sort of check to marker IDs that we do to the map pointer. For the map pointer, we decided to throw an exception if the pointer is null because it shouldn't happen during correct use and there's no sensible way to recover from it. For markers though, it's not unreasonable to accidentally use a marker ID that was previously removed (we return a boolean mostly to express that case). Passing an invalid marker ID into the native code here is an expected circumstance, so let's just do that.